### PR TITLE
feature: Tracetest Linter 🐙

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the first version of the Tracetest Linter to the project, including the following features:
1. Enable/disable the linter
2. Score system
3. Minimum score to break the tests
4. Plugins and rules

## Changes

- Adds new components to support the linter system
- Includes new domain for the BE to support linter plugins and rules
- Wires up the linter to the test runner

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/577fd2cb4f3b45b6be9b8c6c6e0abfe0

### CODE REVIEW

File: server/app/facade.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/kubeshop/tracetest/server/executor"
 	"github.com/kubeshop/tracetest/server/executor/pollingprofile"
 	"github.com/kubeshop/tracetest/server/executor/trigger"
+	linter_resource "github.com/kubeshop/tracetest/server/linter/resource"
```
Comment (danielbdias) [inline server/app/facade.go:10]: Same here. Can we rename this to `camelCase`?

File: server/executor/linter_runner.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/kubeshop/tracetest/server/analytics"
+	"github.com/kubeshop/tracetest/server/linter"
+	linter_resource "github.com/kubeshop/tracetest/server/linter/resource"
```
Comment (danielbdias) [inline server/executor/linter_runner.go:11]: Can we change to `camelCase`?

File: server/executor/linter_runner.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+				e.onError(ctx, request, run, err)
+				return
+			}
+
+			e.onFinish(ctx, request, run)
+		}
+	}
+}
+
+func (e *defaultlinterRunner) Runlinter(ctx context.Context, request LinterRequest) {
```
Comment (danielbdias) [inline server/executor/linter_runner.go:142]:
```suggestion
func (e *defaultlinterRunner) RunLinter(ctx context.Context, request LinterRequest) {
```

**Comment by schoren** [comment]
looks good to me!

File: server/model/events/events.go [hunk 4]
```diff
@@ -415,3 +415,63 @@ func TraceOtlpServerReceivedSpans(testID id.ID, runID, spanCount int, requestTyp
 		Outputs:             []model.OutputInfo{},
 	}
 }
+
+func TracelinterStart(testID id.ID, runID int) model.TestRunEvent {
+	return model.TestRunEvent{
+		TestID:              testID,
+		RunID:               runID,
+		Stage:               model.StageTrace,
+		Type:                "TRACE_linter_START",
```
Comment (danielbdias) [inline server/model/events/events.go:424]:
```suggestion
		Type:                "TRACE_LINTER_START",
```

File: server/model/events/events.go [hunk 5]
```diff
@@ -415,3 +415,63 @@ func TraceOtlpServerReceivedSpans(testID id.ID, runID, spanCount int, requestTyp
...
+		Outputs:             []model.OutputInfo{},
+	}
+}
+
+func TracelinterError(testID id.ID, runID int, err error) model.TestRunEvent {
+	return model.TestRunEvent{
+		TestID:              testID,
+		RunID:               runID,
+		Stage:               model.StageTrace,
+		Type:                "TRACE_linter_ERROR",
```
Comment (danielbdias) [inline server/model/events/events.go:469]:
```suggestion
		Type:                "TRACE_LINTER_ERROR",
```

**Review by danielbdias — APPROVED** [review]
First of all: fantastic work, team! This feature is pretty cool!

I've added some comments about one little thing here or there in the code that we can improve, but that can be on future versions.

File: server/linter/plugins/common/rules/enforce_dns.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/kubeshop/tracetest/server/model"
+)
+
+type ensuresDnsUsage struct {
+	model.BaseRule
+}
+
+var (
+	dnsFields = []string{"net.peer.name", "http.url", "db.connection_string", "net.sock.peer.addr", "net.host.name"}
+)
```
[the remark this answers is not in the document]
Comment (xoscar) [reply server/linter/plugins/common/rules/enforce_dns.go:17]: Yes sir, we are going to have a bunch of fancy ways to configure the octolint

File: web/cypress/e2e/Transactions/TransactionsRun.spec.ts [hunk 7]
```diff
@@ -62,7 +62,7 @@ describe('Transactions', () => {
 
     cy.submitCreateForm('CreateTransactionFactory');
     cy.get('[data-cy=transaction-details-name').should('have.text', `${name} (v1)`);
-    cy.get('[data-cy=transaction-run-button]', {timeout: 30000}).should('be.visible');
+    cy.get('[data-cy=transaction-run-button]', {timeout: 50000}).should('be.visible');
```
[the remark this answers is not in the document]
Comment (xoscar) [reply web/cypress/e2e/Transactions/TransactionsRun.spec.ts:65]: We need to revamp the FE e2e tests, which would include cleaning up all of this 💩 more tests, better tests!

**Pull request merged**